core/sodasql/cli/ingest.py

```diff
@@ -0,0 +1,189 @@
+            expression=test_node.compiled_sql if isinstance(test_node, CompiledSchemaTestNode) else None,
+            metrics=None,
+            column=test_node.column_name,
+        )
+        for index, test_node in enumerate(test_nodes.values())
+    }
+
+    tests_with_test_result = {
+        run_result.unique_id:
+        TestResult(
+            dbt_tests_with_soda_test[run_result.unique_id],
+            passed=run_result.status == TestStatus.Pass,
+            skipped=run_result.status == TestStatus.Skipped,
+            values={"failures": run_result.failures},
+        )
+        for run_result in run_results
+    }
+    return tests_with_test_result
+
+
+def create_dbt_test_results_iterator(
+    manifest_file: Path, run_results_file: Path
+) -> Iterator[tuple[str, list[TestResult]]]:
+    """
+    Create an iterator for the dbt test results.
+
+    Parameters
+    ----------
+    manifest : dict
+        The path to the manifest file.
+    run_results : Path
+        The path to the run results file.
+
+    Returns
+    -------
+    out : Iterator[tuple[str, list[TestResult]]]
+        The table name and its corresponding test results.
+    """
+    try:
+        from sodasql import dbt as soda_dbt
+    except ImportError as e:
+        raise RuntimeError(
+            "Soda SQL dbt extension is not installed: $ pip install soda-sql-dbt"
+        ) from e
+
+    with manifest_file.open("r") as file:
+        manifest = json.load(file)
+    with run_results_file.open("r") as file:
+        run_results = json.load(file)
+
+    model_nodes, test_nodes = soda_dbt.parse_manifest(manifest)
+    parsed_run_results = soda_dbt.parse_run_results(run_results)
+    tests_with_test_result = create_dbt_run_result_to_test_result_mapping(
+        test_nodes, parsed_run_results
+    )
+    models_with_tests = soda_dbt.create_models_to_tests_mapping(
+        model_nodes, test_nodes, parsed_run_results
+    )
+
+    for model_unique_id, test_unique_ids in models_with_tests.items():
+        table_name = model_nodes[model_unique_id].alias
+        test_results = [
+            tests_with_test_result[test_unique_id]
+            for test_unique_id in test_unique_ids
+        ]
+
+        yield table_name, test_results
+
+
+def ingest(
+    tool: str,
+    warehouse_yml_file: str,
+    dbt_manifest: Path | None = None,
+    dbt_run_results: Path | None = None,
+) -> None:
+    """
+    Ingest test information from different tools.
+
+    Arguments
+    ---------
+    tool : str {'dbt'}
+        The tool name.
+    warehouse_yml_file : str
+        The warehouse yml file.
+    dbt_manifest : Optional[Path]
+        The path to the dbt manifest.
+    dbt_run_results : Optional[Path]
+        The path to the dbt run results.
+
+    Raises
+    ------
+    ValueError :
+       If the tool is unrecognized.
+    """
+    logger = logging.getLogger(__name__)
+    logger.info(SODA_SQL_VERSION)
```

Comment:
Add more logging for cli user feedback. @vijaykiran : should we add the telemetry stuff here?

Reply:
I believe we will want to indeed.

Reply:
Then I'll leave it up to you or @vijaykiran to do this, I'm ok with someone making those changes to this PR.